var/spack/repos/builtin/packages/flecsi/package.py

```diff
@@ -33,34 +55,68 @@ class Flecsi(CMakePackage):
     variant('flecstan', default=False,
             description='Build FleCSI Static Analyzer')
 
-    depends_on('cmake@3.1:',  type='build')
+    depends_on('cmake@3.12:',  type='build')
     # Requires cinch > 1.0 due to cinchlog installation issue
     depends_on('cinch@1.01:', type='build')
     depends_on('mpi', when='backend=mpi')
     depends_on('mpi', when='backend=legion')
-    depends_on('gasnet@2019.3.0 ~pshm', when='backend=legion')
-    depends_on('legion@ctrl-rep +shared +mpi', when='backend=legion')
-    depends_on('boost@1.59.0: cxxstd=11 +program_options')
+    depends_on('mpi', when='backend=hpx')
+    depends_on('openmpi@3.1.3:3.1.4', when='+openmpi')
+    depends_on('mpich@3.2.1', when='+mpich')
```

Comment:
I would not add variants for each possible MPI provider. The correct way of handling this in Spack is to request:
```console
$ spack install flecsi backend=mpi ^mpich
```

Reply:
We are trying to achieve that user can pick mpich or openmpi but if they pick through `^mpich` or `^openmpi`, we would like to restrict to only the version we know works.

Reply:
Yep, you can do something like:
```python
conflicts('^mvapich2')
```